File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    longestConsecutive(nums) {
        const obj = new Set(nums);
        nums = nums.map(x => {
            if (!obj.has(x - 1)) {
                console.log('push', x)
                return x;
            }
            return undefined;
        }).filter(x => x!== undefined);

        let h = 0;

        // console.log({nums, obj})
        nums.forEach(x => {
            let hh = 0;
            while(obj.has(x)) {
                x++;
                hh++
                // console.log({x, hh})
            };
            if (h < hh) h = hh;
            // console.log({hh});
        });
        return h;
    }
}

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-0.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        if nums[-1] > nums[0]:
            return nums[0]
        l, r = 0, len(nums) - 1

        min_found = nums[-1]
        while l < r:
            mid = l + (r - l) // 2
            if min_found > nums[mid]:
                min_found = nums[mid]
            if nums[mid] > nums[l]:
                l = mid
            else:
                r = mid
        return min_found

        
        